internal/service/alertconfiguration/model_alert_configuration_test.go

```diff
@@ -0,0 +1,537 @@
+package alertconfiguration_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/alertconfiguration"
+	"go.mongodb.org/atlas-sdk/v20231115002/admin"
+)
+
+const (
+	group               string  = "GROUP"
+	previousIntervalMin int     = 10
+	intervalMin         int     = 5
+	delayMin            int     = 0
+	enabled             bool    = true
+	disabled            bool    = false
+	previousOperator    string  = "MORE_THAN"
+	operator            string  = "LESS_THAN"
+	threshold           float64 = 99.0
+	units               string  = "RAW"
+	mode                string  = "AVERAGE"
+)
+
+func TestNotificationSDKToTFModel(t *testing.T) {
+	testCases := []struct {
+		name                      string
+		SDKResp                   *[]admin.AlertsNotificationRootForGroup
+		currentStateNotifications []alertconfiguration.TfNotificationModel
+		expectedTFModel           []alertconfiguration.TfNotificationModel
+	}{
+		{
+			name: "Complete SDK response",
+			SDKResp: &[]admin.AlertsNotificationRootForGroup{
+				{
+					TypeName:     admin.PtrString(group),
+					IntervalMin:  admin.PtrInt(intervalMin),
+					DelayMin:     admin.PtrInt(delayMin),
+					SmsEnabled:   admin.PtrBool(disabled),
+					EmailEnabled: admin.PtrBool(enabled),
+					ChannelName:  admin.PtrString("#channel"),
+					Roles: []string{
+						"GROUP_DATA_ACCESS_READ_ONLY",
+						"GROUP_CLUSTER_MANAGER",
+						"GROUP_DATA_ACCESS_ADMIN",
+					},
+				},
+			},
+			currentStateNotifications: []alertconfiguration.TfNotificationModel{
+				{
+					TypeName:     types.StringValue(group),
+					IntervalMin:  types.Int64Value(int64(previousIntervalMin)),
+					DelayMin:     types.Int64Value(int64(delayMin)),
+					SMSEnabled:   types.BoolValue(disabled),
+					EmailEnabled: types.BoolValue(enabled),
+					Roles: []string{
+						"GROUP_DATA_ACCESS_READ_ONLY",
+						"GROUP_CLUSTER_MANAGER",
+						"GROUP_DATA_ACCESS_ADMIN",
+					},
+				},
+			},
+			expectedTFModel: []alertconfiguration.TfNotificationModel{
+				{
+					TypeName:     types.StringValue(group),
+					IntervalMin:  types.Int64Value(int64(intervalMin)),
+					DelayMin:     types.Int64Value(int64(delayMin)),
+					SMSEnabled:   types.BoolValue(disabled),
+					EmailEnabled: types.BoolValue(enabled),
+					ChannelName:  types.StringNull(),
```

Comment:
You are right, it does not need to be defined explicitly, but I did it to make explicit the fact that even if in the sdk response `ChannelName` has value, but the current TF state has it null, the final TF model will have it as null because of the 
`if !currState.ChannelName.IsNull() { newState.ChannelName = conversion.StringPtrNullIfEmpty(value.ChannelName)
}`
condition in `NewTFNotificationModelList` method